helix/appendix/alternate-visual-studio-structures.rst

```diff
@@ -0,0 +1,231 @@
+
+How you consolidate the projects in your solution will ultimately depend on
+the issue(s) you are attempting to address, and your view on the purpose and role
+of Visual Studio projects.
+
+* You may group Features into combined projects based on the concept of
+  `Feature Groups </principles/architecture-principles/modules.html#groups>`__. This will
+  reduce the number of projects but care must be taken to avoid dependencies between
+  Features in a group.
+* You may place an entire Layer into its own project as a "logical" grouping of modules.
+  This benefits the Feature Layer most, as it should contain far more modules than other layers.
+  Consolidating the Project Layer likely provides little benefit due to the number of modules, and
+  may create confusion around the composition of each Project Module. You should
+  take care when consolidating the Foundation Layer as well, as the allowance for
+  Foundation-to-Foundation references means that these dependencies need to be closely
+  monitored and managed.
+* You can place all the code for a single deployment target (e.g. main Sitecore website CM/CD code)
+  into a single assembly. This would be the most extreme example of treating a
+  Visual Studio project as a unit of *deployment*. Your boundaries between Sitecore
+  Helix modules becomes entirely logical, and Common Closure is lost to a large degree.
+* You can split out modules as needed for reuse. This is not explicitly advocated by Sitecore Helix
+  but could be a requirement in organizations with multiple Sitecore deployments and a common
+  shared code base. However in this case you likely should be independently managing
+  and versioning these modules, and distributing them to other solutions via NuGet.
+
+In all of the approaches above, you will need to consider the structure of your tests,
+serialized items, and projects for `additional Sitecore services </principles/services/index>`__
+as well. At the very least you should aim to retain the "logical" boundaries betweeen modules
+by mirroring the filesystem and namespace structure within each of these.
+
+It's strongly recommended that you use tools such as `FxCop rules <https://www.teamdevelopmentforsitecore.com/Blog/sitecore-helix-fxcop-rules>`__
+and `Sitecore TDS Validation <https://www.hhogdev.com/help/tds/propvalidation>`__ to enforce
+dependency rules between your modules if you consolidate Visual Studio projects in any
+way.
+
+
+Does Sitecore Helix really allow this?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Sitecore Helix is not a rulebook </introduction/about-this-documentation.html#sitecore-helix-is-not-a-rulebook>`__.
+You should be pragmatic about applying its conventions rather than religious. When
+deviating from them however, you should always:
+
+* Understand why, understand the positive and negative impact of doing so,
+  and anticipate the technical debt you may be accumulating.
+* Document the deviation and alternative convention(s).
```

Comment:
💯

Reply:
Thanks to @derekcorreia for noting that we needed to call that out